zokrates_analysis/src/reducer/inline.rs

```diff
@@ -58,35 +57,40 @@ pub enum InlineError<'ast, T> {
 }
 
 fn get_canonical_function<'ast, T: Field>(
-    function_key: DeclarationFunctionKey<'ast, T>,
+    function_key: &DeclarationFunctionKey<'ast, T>,
     program: &TypedProgram<'ast, T>,
 ) -> TypedFunctionSymbolDeclaration<'ast, T> {
     let s = program
         .modules
         .get(&function_key.module)
         .unwrap()
         .functions_iter()
-        .find(|d| d.key == function_key)
+        .find(|d| d.key == *function_key)
         .unwrap();
 
     match &s.symbol {
-        TypedFunctionSymbol::There(key) => get_canonical_function(key.clone(), program),
+        TypedFunctionSymbol::There(key) => get_canonical_function(key, program),
         _ => s.clone(),
     }
 }
 
 type InlineResult<'ast, T> = Result<
-    Output<(Vec<TypedStatement<'ast, T>>, TypedExpression<'ast, T>), Vec<Versions<'ast>>>,
+    (
```

Comment:
I think we should have a struct for this, it's a bit unclear by just looking at this type what the elements represent